fdbserver/DDTeamCollection.actor.cpp

```diff
@@ -1442,6 +1442,20 @@ class DDTeamCollectionImpl {
 					when(wait(storageMetadataTracker || storeTypeTracker)) {}
 					when(wait(server->ssVersionTooFarBehind.onChange())) {}
 					when(wait(self->disableFailingLaggingServers.onChange())) {}
+					when(wait(server->storageQueueTooLong.onChange())) {
+						TraceEvent(SevInfo, "StorageQueueTooLongTriggerSplit", self->distributorId)
+						    .detail("SSID", server->getId());
+						std::vector<ShardsAffectedByTeamFailure::Team> teams;
+						for (const auto& team : server->getTeams()) {
+							std::vector<UID> servers;
+							for (const auto& server : team->getServers()) {
+								servers.push_back(server->getId());
+							}
+							teams.push_back(ShardsAffectedByTeamFailure::Team(servers, self->primary));
+						}
+						self->triggerSplitForStorageQueueTooLong.send(
```

Comment:
Sorry for the confusing. Here, we send all teams of the server to DDtracker. However, DDtracker will only choose one team to move a shard out of the team. This shard is the hottest shard among all shards of all teams of this server. The reason we have to send all teams to DDtracker is because ServerTracker does not have shard metrics information, whereas DDtracker has the metrics.